### ISSUE TITLE
lwip build from source?

### ISSUE THREAD

**Issue by eastein** [pfalcon/esp-open-sdk#69]
Currently liblwip.a is being taken direct from the espressif SDK binary libs, correct? Have you explored the work done by @kadamski at https://github.com/kadamski/esp-lwip - is there a reason not to use this?

I suppose since liblwip source is not being regularly released by Espressif this might be a problem - this was based on a dump of changes on top of 1.4.0.

**Comment by pfalcon** [comment]
Yes, sure, doing it like that was the idea behind esp-open-sdk, except that I cannot do it all myself. @kadamski's work is nice and good progress towards that aim, but:

1. His project isn't up to date wrt to recent SDKs - not only latest 1.2.0, but even 1.0.1: https://github.com/kadamski/esp-lwip/issues/7
2. @kadamski himself isn't very responsive.

I opened https://github.com/kadamski/esp-lwip/issues/9 trying to discuss how to ease maintenance of his lwIP port, but that all appears to be blocked on time availability.

Summing up: if you have time/interest, feel free to start on that and keep in touch.

**Comment by eastein** [comment]
Yeah, I kind of figured this would be the reason. I wanted to make sure there wasn't a "Well, this is never going to work well for x reasons" reason to not do it before I considered if I wanted to do it.

**Comment by eastein** [comment]
It seems to be a pretty huge task to make that work for all SDK versions, or even any recent ones - and the amount of testing it has received has so far been low.

**Comment by eastein** [comment]
Thanks for the info; perhaps at some point I will have the energy for it.

**Comment by pfalcon** [comment]
> "Well, this is never going to work well for x reasons"

Nope, main concerns are "it would be nice, and can be done, but I don't need it that much (I'm satisfied with 'can be done' assessment I'm performed)" and "esp8266 is just a random chip, popping out of nowhere several months ago and likely going back in some time". Note that various people doing these things in different ways, there's for example https://github.com/SuperHouse/esp-open-rtos , which is on a way to build lwIP, axTLS, etc. from source - but in FreeRTOS environment.

**Comment by eastein** [comment]
Ah nice, thanks for the esp-open-rtos pointer.

**Comment by stv0g** [comment]
Espresso recently released their version of LwIP in source format:
http://bbs.espressif.com/viewtopic.php?f=46&t=951

This should allow us to build it at least for the newest (1.3.0) SDK version.
I stil have to test, weather their own SDK properly links against a manually compiled LwIP.

**Comment by eastein** [comment]
oooh nice

**Comment by eastein** [comment]
I care about this because the built in stack only supports A records for DNS, and I want SRV. So I've been implementing my own SRV queryer with a little bit of code taken from LWIP. I'd like to contribute it back possibly by integrating it in, although I'm not sure the LWIP folks care about it.

**Comment by jrmuizel** [comment]
Is this resolved by https://github.com/pfalcon/esp-open-lwip?

**Comment by pfalcon** [comment]
Yes, https://github.com/pfalcon/esp-open-lwip largely resolved this. And the final step is https://github.com/pfalcon/esp-open-sdk/issues/263
